cmd/kubeadm/app/constants/constants_test.go

```diff
@@ -275,3 +275,75 @@ func TestGetSkewedKubernetesVersionImpl(t *testing.T) {
 		})
 	}
 }
+
+func TestGetAPIServerVirtualIP(t *testing.T) {
+	var tests = []struct {
+		name, svcSubnet, expectedIP string
+		expectedErr                 bool
+	}{
+		{
+			name:        "subnet mask 12",
+			svcSubnet:   "10.96.0.0/12",
+			expectedIP:  "10.96.0.1",
+			expectedErr: false,
+		},
+		{
+			name:        "subnet mask 26",
+			svcSubnet:   "10.87.116.64/26",
+			expectedIP:  "10.87.116.65",
+			expectedErr: false,
+		},
+		{
+			name:        "dual-stack ipv4 primary, subnet mask 26",
+			svcSubnet:   "10.87.116.64/26,fd03::/112",
+			expectedIP:  "10.87.116.65",
+			expectedErr: false,
+		},
+		{
+			name:        "dual-stack, subnet mask 26 , missing first ip segment",
+			svcSubnet:   ",10.87.116.64/26",
+			expectedErr: true,
+		},
+		{
+			name:        "dual-stack ipv4 primary, subnet mask 26, missing second ip segment",
+			svcSubnet:   "10.87.116.64/26,",
+			expectedErr: true,
+		},
+		{
+			name:        "dual-stack ipv6 primary, subnet mask 112",
+			svcSubnet:   "fd03::/112,10.87.116.64/26",
+			expectedIP:  "fd03::1",
+			expectedErr: false,
+		},
+		{
+			name:        "dual-stack, subnet mask 26, missing first ip segment",
+			svcSubnet:   ",fd03::/112",
+			expectedErr: true,
+		},
+		{
+			name:        "dual-stack, subnet mask 26, missing second ip segment",
+			svcSubnet:   "fd03::/112,",
+			expectedErr: true,
+		},
+	}
+	for _, rt := range tests {
+		t.Run(rt.name, func(t *testing.T) {
+			virtualIP, err := GetAPIServerVirtualIP(rt.svcSubnet)
+			if rt.expectedErr {
+				if err == nil {
+					t.Errorf("failed APIServerVirtualIP:\n\texpected error, but got no error")
+				}
+			} else if !rt.expectedErr && err != nil {
```

Comment:
these branches can be
```
if (err != nil) != rt.expectedError {
...

```

Reply:
I have modified as you suggested, thank you for your review